lib/awful/widget/calendar.lua

```diff
@@ -0,0 +1,360 @@
+    --                        cl, cc, cr
+    --                        bl, bc, br
+    local x,y
+    if pos:sub(1,1) == "t" then
+        y = wa.y
+    elseif pos:sub(1,1) == "b" then
+        y = wa.y + wa.height - height
+    else  --if pos:sub(1,1) == "c" then
+        y = wa.y + math.floor((wa.height - height) / 2)
+    end
+    if pos:sub(2,2) == "l" then
+        x = wa.x
+    elseif pos:sub(2,2) == "r" then
+        x = wa.x + wa.width - width
+    else  --if pos:sub(2,2) == "c" then
+        x = wa.x + math.floor((wa.width - width) / 2)
+    end
+
+    return {x=x, y=y, width=width, height=height}
+end
+
+--- Call the calendar with offset
+-- @tparam number inc_offset Offset with respect to current date
+-- @tparam table props Properties of the calendar (screen, position, ...)
+-- @treturn wibox The wibox calendar
+function calendar_popup:call_calendar(inc_offset, props)
+    local props = props or {}
+    local offset, position, s = inc_offset or 0, props.position or self.position, props.screen or self.screen or ascreen.focused()
+    self.position = position  -- remember last position when changing offset
+
+    self.offset = self.offset + offset
+    local is_current = (offset == 0 or self.offset == 0)
+    if is_current then
+        self.offset = 0
+    end
+
+    local widget = self:get_widget()
+    local date, hint_width, hint_height
+    if widget.type == "month" then
+        if is_current then
+            date = os.date("%d %m %Y")
+        else
+            -- calculate the offset date
+            local month = tonumber(os.date("%m"))
+            local year  = tonumber(os.date("%Y"))
+
+            month = month + self.offset
+
+            while month > 12 do
+                month = month - 12
+                year = year + 1
+            end
+
+            while month < 1 do
+                month = month + 12
+                year = year - 1
+            end
+
+            date = string.format("%s %s", month, year)
+        end
+    elseif widget.type == "year" then
+        date = string.format("%s", tonumber(os.date("%Y")) + self.offset)
+    end
+
+    -- set date and screen before updating geometry
+    widget:set_date(date)
+    self:set_screen(s)
+    -- update geometry (depends on date and screen)
+    self:geometry(get_geometry(widget, s, position))
+    return self
+end
+
+
+--- Toggle calendar visibility
+function calendar_popup:toggle()
+    self:call_calendar(0)
+    self.visible = not self.visible
+end
+
+
+--- Attach the calendar to a widget to display at a specific position.
+--
+--    local mytextclock = wibox.widget.textclock()
+--    local month_calendar = calendar.month()
+--    month_calendar:attach(mytextclock, 'tr')
+--
+-- @param widget Widget to attach the calendar
+-- @tparam[opt="tr"] string position Two characters string defining the position on the screen
+-- @treturn wibox The wibox calendar
+function calendar_popup:attach(widget, position)
+    local position = position or "tr"
```

Comment:
luacheck is complaining about this, forgot what the fix was.

    lib/awful/widget/calendar.lua:271:11: variable position was previously defined as an argument on line 270